pyccel/ast/numpyext.py

```diff
@@ -1277,6 +1278,15 @@ class NumpyArctanh(NumpyUfuncUnary):
 
 #=======================================================================================
 
+
+class NumpySign(NumpyUfuncUnary):
+    """Represent a call to the sign function in the Numpy library"""
+    __slots__ = ()
+    name = 'sign'
+    def _set_dtype_precision(self, x):
+        self._dtype     = x.dtype
+        self._precision = default_precision[str_dtype(self._dtype)]
```

Comment:
Are you sure that the precision should always be equal to the default precision? Similar code is used in lots of functions in this file, but I suspect these are bugs (similar to #1177 )